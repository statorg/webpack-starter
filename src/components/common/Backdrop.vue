<template>
  <transition
    enter-class="backdrop_out"
    enter-active-class=""
    enter-to-class=""
    leave-class=""
    leave-active-class=""
    leave-to-class="backdrop_out"
  >
    <div
      v-if="visible"
      class="backdrop"
      v-on="$listeners"
    />
  </transition>
</template>

<script>
export default {
  name: 'Backdrop',
  props: {
    visible: {
      type: Boolean
    }
  }
}
</script>

<style lang="scss">
$backdrop-transition: 0.3s easing(swing);

.backdrop {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: z(backdrop);
  background-color: #212121;
  opacity: 0.46;
  /*transition: opacity 0.4s cubic-bezier(0.35, 0, 0.25, 1);*/
  transition: opacity $backdrop-transition;
  will-change: opacity;

  &_out {
    opacity: 0;
  }
}
</style>
