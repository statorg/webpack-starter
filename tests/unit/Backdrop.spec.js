import { shallowMount } from '@vue/test-utils'
import Backdrop from '@/components/common/Backdrop'

describe('ToTop.vue', () => {
  it('should not render when visible prop is not passed', () => {
    const wrapper = shallowMount(Backdrop)
    expect(wrapper.isEmpty()).toBe(true)
  })

  it('should render when visible prop is passed as true', () => {
    const wrapper = shallowMount(Backdrop, {
      propsData: {
        visible: true
      }
    })
    expect(wrapper.isEmpty()).toBe(false)
  })

  it('should emit click event when clicked', () => {
    const onClick = jest.fn()
    const wrapper = shallowMount(Backdrop, {
      propsData: {
        visible: true
      },
      listeners: {
        click: onClick
      }
    })
    wrapper.trigger('click')
    expect(onClick).toHaveBeenCalled()
    expect(onClick.mock.calls[0][0]).toBeInstanceOf(MouseEvent)
  })

  it('should render and match snapshot', () => {
    const wrapper = shallowMount(Backdrop)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
