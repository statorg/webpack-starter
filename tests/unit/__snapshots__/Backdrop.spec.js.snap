// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToTop.vue should render and match snapshot 1`] = `undefined`;
